cms/djangoapps/contentstore/rest_api/v0/views/details_settings.py

```diff
@@ -0,0 +1,39 @@
+""" API Views for course details settings """
+
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.request import Request
+from rest_framework.views import APIView
+from xmodule.modulestore.django import modulestore
+
+from cms.djangoapps.models.settings.encoder import CourseSettingsEncoder
+from common.djangoapps.student.auth import has_studio_read_access, has_studio_write_access
+from common.djangoapps.util.json_request import JsonResponse, expect_json
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin, verify_course_exists, view_auth_classes
+from openedx.core.djangoapps.models.course_details import CourseDetails
+
+from ....views.course import update_course_details_settings
+
+
+@view_auth_classes(is_authenticated=True)
+@expect_json
```

Comment:
Is this required? DRF should support JSON natively.

Reply:
It was needed mostly to load the data into `request.json` object because the common method `update_course_details_settings` uses that. Without this decorator request object won't have the `json` attribute.